GeneratorInterface/GenFilters/src/pMSSMFilter.cc

```diff
@@ -0,0 +1,166 @@
+/*
+Package:    GeneralInterface/GenFilters/pMSSMFilter
+Class:      pMSSMFilter
+
+class pMSSMFilter pMSSMFilter.cc GeneratorInterface/GenFilters/src/pMSSMFilter.cc
+
+Description: EDFilter which checks the event passes a baseline selection for the run-II pMSSM effort.
+
+Implementation: 
+
+The following input parameters are used 
+  gpssrc = cms.InputTag("X") : gen particle collection label as input
+  jetsrc  = cms.InputTag("X") : genjet collection collection label as input
+  jetPtCut = cms.double(#) : GenJet pT cut for HT
+  jetEtaCut = cms.double(#) : GenJet eta cut for HT
+  genHTcut = cms.double(#) : GenHT cut
+  muPtCut = cms.double(#) : muon pT cut
+  muEtaCut = cms.double(#) : muon eta cut
+  elPtCut = cms.double(#) : electron pT cut
+  elEtaCut = cms.double(#) : electron eta cut
+  gammaPtCut = cms.double(#) : photon pT cut
+  gammaEtaCut = cms.double(#) : photon eta cut
+  loosemuPtCut = cms.double(#) : loose muon pT cut
+  looseelPtCut = cms.double(#) : loose electron pT cut
+  loosegammaPtCut = cms.double(#) : loose photon pT cut
+  veryloosegammaPtCut = cms.double(#) : even looser photon pT cut
+Original Author:  Malte Mrowietz
+         Created:  Jun 2019
+*/
+
+//System include files
+#include <cmath>
+#include <memory>
+#include <vector>
+//User include files
+#include "FWCore/Framework/interface/EDFilter.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
+#include "DataFormats/JetReco/interface/GenJetCollection.h"
+
+//Class declaration
+class pMSSMFilter : public edm::EDFilter {
```

Comment:
Could this be made a non-legacy module? By quick look it could be made to a global module pretty easily.

Reply:
> 
> 
> Could this be made a non-legacy module? By quick look it could be made to a global module pretty easily.

I am not sure exactly what that implies. We need it for the non-legacy re-reco campaign where we are using CMSSW_9_4_X. As long as the backport is possible, I am open to it.

Reply:
Thread-friendly module types have been available since LS1, so a backport to 9_4_X should be as straightforward as with a legacy module type.